components/brave_wallet_ui/common/hooks/assets-management.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { SimpleActionCreator, EmptyActionCreator } from 'redux-act'
+import {
+  BraveWallet
+} from '../../constants/types'
+
+import {
+  AddUserAssetPayloadType,
+  SetUserAssetVisiblePayloadType,
+  RemoveUserAssetPayloadType
+} from '../constants/action_types'
+
+import { stripERC20TokenImageURL } from '../../utils/string-utils'
+
+export default function useAssetManagement (
+  addUserAsset: SimpleActionCreator<AddUserAssetPayloadType>,
+  setUserAssetVisible: SimpleActionCreator<SetUserAssetVisiblePayloadType>,
+  removeUserAsset: SimpleActionCreator<RemoveUserAssetPayloadType>,
+  refreshBalancesPricesAndHistory: EmptyActionCreator,
+  selectedNetwork: BraveWallet.EthereumChain,
+  fullTokenList: BraveWallet.BlockchainToken[],
+  userVisibleTokensInfo: BraveWallet.BlockchainToken[]
+) {
+  const onAddUserAsset = (token: BraveWallet.BlockchainToken) => {
+    addUserAsset({
+      token: {
+        ...token,
+        logo: stripERC20TokenImageURL(token.logo) || ''
+      },
+      chainId: selectedNetwork.chainId
+    })
+  }
+
+  const onAddCustomAsset = (token: BraveWallet.BlockchainToken) => {
+    onAddUserAsset(token)
+    refreshBalancesPricesAndHistory()
+  }
+
+  const findVisibleTokenInfo = (token: BraveWallet.BlockchainToken) => {
+    return userVisibleTokensInfo.find((t) => t.contractAddress === token.contractAddress)
+  }
+
+  const onUpdateVisibleAssets = React.useCallback((updatedTokensList: BraveWallet.BlockchainToken[]) => {
+    // Gets a list of all added tokens and adds them to the userVisibleTokensInfo list
+    const addedTokens = updatedTokensList.filter((firstItem) => !userVisibleTokensInfo.some((secondItem) => firstItem.contractAddress === secondItem.contractAddress))
+    if (addedTokens.length !== 0) {
+      addedTokens.forEach((token) => onAddUserAsset(token))
+    }
+
+    // Gets a list of all removed tokens and removes them from the userVisibleTokensInfo list
+    const removedTokens = userVisibleTokensInfo.filter((firstItem) => !updatedTokensList.some((secondItem) => firstItem.contractAddress === secondItem.contractAddress))
+    if (removedTokens.length !== 0) {
+      removedTokens.forEach((token) => removeUserAsset({ token, chainId: selectedNetwork.chainId }))
+    }
+
+    // Gets a list of custom tokens returned from updatedTokensList payload
+    const customTokens = updatedTokensList.filter((firstItem) =>
+      !fullTokenList.some((secondItem) => firstItem.contractAddress === secondItem.contractAddress))
+
+    // Compares customTokens against userVisibleTokensInfo list and updates the custom tokens visibility if it has changed
+    if (customTokens.length !== 0) {
```

Comment:
This `if` condition is not necessary, since `forEach` block will be skipped for empty list.

Reply:
Removed!